sdk/core/core/src/az_log.c

```diff
@@ -176,6 +176,28 @@ static az_result _az_log_http_response_msg(
   return _az_log_http_request_msg(log_msg_bldr, hrb, 1);
 }
 
+static az_result _az_log_http_retry_msg(
+    az_span_builder * const log_msg_bldr,
+    int16_t const attempt,
+    int32_t const delay_msec) {
+  AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR("HTTP Retry attempt")));
+
+  if (attempt >= 0) {
+    AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR(" #")));
+    AZ_RETURN_IF_FAILED(az_span_builder_append_uint64(log_msg_bldr, (uint64_t)attempt));
+  }
+
+  AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR(" will be made")));
+
+  if (delay_msec >= 0) {
+    AZ_RETURN_IF_FAILED(az_span_builder_append(log_msg_bldr, AZ_STR(" in ")));
+    AZ_RETURN_IF_FAILED(az_span_builder_append_uint64(log_msg_bldr, (uint64_t)delay_msec));
```

Comment:
We have a brand new append_double method now in the new az_span.h file. Use this instead of casting the double to a uint64_t.

Reply:
Will do; but this is not double.

Reply:
I'll also see if I can improve this code so that the cast is not necessary. Currently, it is technically not required here, but it would be required if that was a negative value.